server/models/helpers/SearchHelper.ts

```diff
@@ -354,36 +352,57 @@ export default class SearchHelper {
       });
     }
 
-    if (!options.includeArchived) {
-      where[Op.and].push({
-        archivedAt: {
-          [Op.eq]: null,
-        },
-      });
-    }
-
-    if (options.includeDrafts && model instanceof User) {
-      where[Op.and].push({
-        [Op.or]: [
+    const statusQuery = [];
+    if (options.statusFilter?.includes(StatusFilter.Published)) {
+      statusQuery.push({
+        [Op.and]: [
           {
             publishedAt: {
               [Op.ne]: null,
             },
+            archivedAt: {
+              [Op.eq]: null,
+            },
           },
+        ],
+      });
+    }
+
+    if (options.statusFilter?.includes(StatusFilter.Draft)) {
+      statusQuery.push({
+        [Op.and]: [
           {
-            createdById: model.id,
+            publishedAt: {
+              [Op.eq]: null,
+            },
+            archivedAt: {
+              [Op.eq]: null,
+            },
+            ...(model instanceof User && {
+              [Op.or]: [
+                { createdById: model.id },
+                { "$memberships.id$": { [Op.ne]: null } },
+              ],
+            }),
           },
-          { "$memberships.id$": { [Op.ne]: null } },
         ],
       });
-    } else {
-      where[Op.and].push({
-        publishedAt: {
+    }
+
+    if (options.statusFilter?.includes(StatusFilter.Archived)) {
+      statusQuery.push({
+        archivedAt: {
           [Op.ne]: null,
         },
       });
     }
 
+    if (options.statusFilter?.length) {
```

Comment:
Previously archived documents were filtered by default, we should retain this behavior. Given the new changes it probably makes sense to do this on the frontend by making `published,draft` statusFilter enabled in the client by default.